internal/api/grpc/server/middleware/auth_interceptor.go

```diff
@@ -47,11 +42,38 @@ func authorize(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo,
 	return handler(ctxSetter(ctx), req)
 }
 
+func orgIDAndDomainFromRequest(ctx context.Context, req interface{}) (id, domain string) {
+	orgID := grpc_util.GetHeader(ctx, http.ZitadelOrgID)
+	o, ok := req.(OrganizationFromRequest)
+	if !ok {
+		return orgID, ""
+	}
+	id = o.OrganizationFromRequest().ID
+	domain = o.OrganizationFromRequest().Domain
+	if id != "" || domain != "" {
+		return id, domain
+	}
+	// check if the deprecated organisation is used.
+	// to be removed before going GA (https://github.com/zitadel/zitadel/issues/6718)
+	id = o.OrganisationFromRequest().ID
+	domain = o.OrganisationFromRequest().Domain
+	if id != "" || domain != "" {
+		return id, domain
+	}
+	return orgID, domain
+}
```

Comment:
As `OrganizationFromRequest()` / `OrganisationFromRequest()` return a pointer `*Organizaion`, shouldn't there be a `nil` check here? Even better, can't we just check for the existence of one or the other based on `nil` check then return the orgID if it was present:


```suggestion
	org := o.OrganizationFromRequest()
	// check if the deprecated organisation is used.
	// to be removed before going GA (https://github.com/zitadel/zitadel/issues/6718)
	if org == nil {
		org = o.OrganisationFromRequest()
	}
	if org != nil {
		if org.ID != "" {
			return org.ID, org.Domain
		}
		return orgID, org.Domain
	}
	return orgID, ""
```

Reply:
I see now that the implementation of the above methods actually never return `nil`. So not sure if the comment is still valid.

Best preactice would be a `nil` check. But then the implementation should also return a `nil` for either case if it wasn't set.